src/libraries/System.Formats.Tar/src/System/Formats/Tar/TarWriter.Unix.cs

```diff
@@ -60,12 +64,31 @@ public sealed partial class TarWriter : IDisposable
 
             entry._header._mode = (status.Mode & 4095); // First 12 bits
 
-            entry.Uid = (int)status.Uid;
-            entry.Gid = (int)status.Gid;
+            // Uid and UName
+            entry._header._uid = (int)status.Uid;
+            _userIdentifiers ??= new Dictionary<uint, string>();
+            if (!_userIdentifiers.ContainsKey(status.Uid))
+            {
+                entry._header._uName = Interop.Sys.GetUName(status.Uid) ?? string.Empty;
+                _userIdentifiers.Add(status.Uid, entry._header._uName);
+            }
+            else
+            {
+                entry._header._uName = _userIdentifiers[status.Uid];
+            }
```

Comment:
We can use TryGetValue instead of ContainsKey + indexer access.

```suggestion
            if (!_userIdentifiers.TryGetValue(status.Uid, out string? userName))
            {
                userName = Interop.Sys.GetUName(status.Uid) ?? "";
                _userIdentifiers.Add(status.Uid, userName);
            }
            entry._header._uName = userName;
```